packages/bot/src/stores/run-panel-store.js

```diff
@@ -54,8 +56,15 @@ export default class RunPanelStore {
             this.is_drawer_open = true;
         }
 
+        this.root_store.contract_card.is_loading = true;
         this.is_run_button_clicked = true;
         Blockly.BLOCKLY_CLASS_OLD.run();
+        this.setContractStage(CONTRACT_STAGES.purchase_sent);
+    }
+
+    @action.bound
+    setContractStage(contract_stage) {
```

Comment:
We don't need this setter, we can just update the value `this.contract_stage = "value";`

Reply:
the value we're setting is an object, so `this.contract_stage = observable(value)`